### PR TITLE
feat: Add 'Recasting of IRIS-HEP Analysis Grand Challenge' project

### PR DESCRIPTION
**Pull request by matthewfeickert**
* Add project for recasting the AGC.
* Note that the 'CMS RECAST Example' can be done on CMS open data.

### CODE REVIEW

File: projects/agc-recast.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+name: Recasting of IRIS-HEP Analysis Grand Challenge
```
Comment (matthewfeickert) [inline projects/agc-recast.yml:2]: @cranmer @lukasheinrich Throwing up a quick draft version now. Happy to revise and clarify any text.

**Comment by matthewfeickert** [comment]
@davidlange6 Can you comment on how we can build the http://research-software-collaborations.org/projects page is actually getting built and recommend a workflow for building it locally to preview changes?

**Comment by davidlange6** [comment]
it gets pulled in as a submodule to https://github.com/research-software-collaborations/research-software-collaborations.github.io, whose readme has some instructions. I see I never connected that information into the README of this repo. will fix that.

**Comment by matthewfeickert** [comment]
@cranmer @lukasheinrich This is going to get merged so it will show up, but we can iterate on the text more later.

**Comment by matthewfeickert** [comment]
@davidlange6 this is ready for review when you have time.

**Pull request merged**